module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -102,4 +157,213 @@ public function listAction()
+                throw new \VuFind\Exception\BadRequest(
+                    'error_inconsistent_parameters'
+                );
+            }
+
+            $updateFields = $this->getUpdateFieldsFromGatheredDetails(
+                $holdConfig,
+                $gatheredDetails,
+                $pickupLocations
+            );
+            if ($updateFields) {
+                $results
+                    = $catalog->updateHolds($selectedIds, $updateFields, $patron);
+                $successful = 0;
+                $failed = 0;
+                foreach ($results as $result) {
+                    if ($result['success']) {
+                        ++$successful;
+                    } else {
+                        ++$failed;
+                    }
+                }
+                // Store results in the session so that they can be displayed when
+                // the user is redirected back to the holds list:
+                $this->getHoldUpdateResultsContainer()->results = $results;
+                if ($successful) {
+                    $msg = $this->translate(
+                        'hold_edit_success_items',
+                        ['%%count%%' => $successful]
+                    );
+                    $this->flashMessenger()->addSuccessMessage($msg);
+                }
+                if ($failed) {
+                    $msg = $this->translate(
+                        'hold_edit_failed_items',
+                        ['%%count%%' => $failed]
+                    );
+                    $this->flashMessenger()->addErrorMessage($msg);
+                }
+                return $this->inLightbox()
+                    ? $this->getRefreshResponse()
+                    : $this->redirect()->toRoute('holds-list');
+            }
+        }
+
+        $view = $this->createViewModel(
+            [
+                'selectedIDS' => $selectedIds,
+                'fields' => $holdConfig['updateFields'],
+                'gatheredDetails' => $gatheredDetails,
+                'pickupLocations' => $pickupLocations,
+            ]
+        );
+
+        return $view;
+    }
+
+    /**
+     * Get list of pickup locations based on the first selected hold. This may not be
+     * perfect as pickup locations may differ per hold, but it's the best we can do.
+     *
+     * @param array $patron      Patron information
+     * @param array $selectedIds Selected holds
+     *
+     * @return array
+     */
+    protected function getPickupLocationsForEdit(array $patron, array $selectedIds
+    ): array {
+        $catalog = $this->getILS();
+        $holds = $catalog->getMyHolds($patron);
+        $firstDetails = reset($selectedIds);
+        $pickupLocations = [];
+        foreach ($holds as $hold) {
+            if ((string)($hold['updateDetails'] ?? '') === $firstDetails) {
+                try {
+                    $pickupLocations = $catalog->getPickUpLocations($patron, $hold);
```

Comment:
Since this is a method now, I think you can slightly simplify things by just directly returning here, eliminating the break and the $pickupLocations variable, and returning an empty array at the bottom of the function as a default. I think that's functionally equivalent but slightly simpler.

Reply:
Simplified. I decided to return default pickup locations at the bottom as a last resort, though I'm not quite sure if that's better than not providing the choice in the first place.